tree/dataframe/test/dataframe_snapshot.cxx

```diff
@@ -1455,4 +1455,108 @@ TEST(RDFSnapshotMore, ZeroOutputEntriesMT)
    gSystem->Unlink(fname);
 }
 
+// Test for custom basket size in Snapshot
+class SnapshotCustomBasketRAII {
+private:
+   std::string fInputFile;
+   std::string fOutputFileCustom;
+   std::string fOutputFileCollection;
+   std::unique_ptr<TFile> fFile;
+   std::unique_ptr<TTree> fTree;
+
+public:
+   SnapshotCustomBasketRAII() : 
+      fInputFile("input_file.root"),
+      fOutputFileCustom("output_file_custom_basket.root"),
+      fOutputFileCollection("output_file_collection_basket.root")
+   {
+      // Create input file and tree
+      fFile.reset(TFile::Open(fInputFile.c_str(), "RECREATE"));
+      fTree.reset(new TTree("tree", "Test Tree"));
+
+      // Create scalar branch
+      float value = 0;
+      fTree->Branch("branch_x", &value);
+
+      // Create vector branch
+      std::vector<float> vec_values;
+      auto vec_branch = fTree->Branch("branch_vec", &vec_values);
+
+      // Fill the tree with some data
+      for (int i = 0; i < 1000; ++i) {
+         value = i;
+         vec_values.clear();
+         // Add some random number of elements to the vector
+         int vec_size = i % 10 + 1;  // 1 to 10 elements
+         for (int j = 0; j < vec_size; ++j) {
+            vec_values.push_back(i + j * 0.1f);
+         }
+         fTree->Fill();
+      }
+      fFile->Write();
+   }
+
+   ~SnapshotCustomBasketRAII() {
+      if (fFile) {
+         fFile->Close();
+      }
```

Comment:
If you create the TFile on the stack in the constructor there will be no need for this extra logic